client.go

```diff
@@ -0,0 +1,268 @@
+// Copyright 2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcreflect
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/bufbuild/connect-go"
+	reflectionv1 "github.com/bufbuild/connect-grpcreflect-go/internal/gen/go/connectext/grpc/reflection/v1"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/descriptorpb"
+)
+
+func NewClientV1(httpClient connect.HTTPClient, baseURL string, options ...connect.ClientOption) *Client {
+	return newClient(
+		httpClient,
+		baseURL+serviceNameV1,
+		options...,
+	)
+}
+
+func NewClientV1Alpha(httpClient connect.HTTPClient, baseURL string, options ...connect.ClientOption) *Client {
+	return newClient(
+		httpClient,
+		baseURL+serviceNameV1Alpha,
+		options...,
+	)
+}
+
+func newClient(httpClient connect.HTTPClient, serviceURL string, options ...connect.ClientOption) *Client {
+	client := connect.NewClient[reflectionv1.ServerReflectionRequest, reflectionv1.ServerReflectionResponse](
+		httpClient,
+		serviceURL+methodName,
+		options...,
+	)
+	return &Client{client: client}
+}
+
+type reflectClient = connect.Client[reflectionv1.ServerReflectionRequest, reflectionv1.ServerReflectionResponse]
+type reflectStream = connect.BidiStreamForClient[reflectionv1.ServerReflectionRequest, reflectionv1.ServerReflectionResponse]
+
+type Client struct {
+	client *reflectClient
+}
+
+type ClientStreamOption func(*clientStreamOptions)
+
+func WithRequestHeaders(headers http.Header) ClientStreamOption {
+	return func(o *clientStreamOptions) {
+		o.headers = headers
+	}
+}
+
+func WithReflectionHost(host string) ClientStreamOption {
+	return func(o *clientStreamOptions) {
+		o.host = host
+	}
+}
+
+type clientStreamOptions struct {
+	host    string
+	headers http.Header
+}
+
+func (c *Client) NewReflectionStream(ctx context.Context, options ...ClientStreamOption) *ClientStream {
```

Comment:
Possibly just `NewStream`? It's already a reflection client.

Reply:
Done